### PR TITLE
Add azurecommons impl.

### PR DESCRIPTION
**Pull request by lafernando**

### CODE REVIEW

File: azurecommons/azurecommons.bal [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+# Generates canonicalized header string from the given header map.
+#
+# + headers - The header map
+# + return - The canonicalized header string
+public function generateCanonicalizedHeadersString(map<string> headers) returns string {
+    // TODO: sort headers - when sort functionality is availble
+    string result = "";
+    foreach var (key, value) in headers {
+      if (key.indexOf("x-ms-") == 0) {
```
Comment (Shan1024) [inline azurecommons/azurecommons.bal]:
```suggestion
        if (key.indexOf("x-ms-") == 0) {
```

File: azurecommons/azurecommons.bal [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          result = result + key.toLower().trim() + ":" + value.trim() + "\n";
```
Comment (Shan1024) [inline azurecommons/azurecommons.bal]:
```suggestion
            result = result + key.toLower().trim() + ":" + value.trim() + "\n";
```

File: azurecommons/azurecommons.bal [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      }
```
Comment (Shan1024) [inline azurecommons/azurecommons.bal]:
```suggestion
        }
```

File: azurecommons/azurecommons.bal [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    }
+    return result;
+}
+
+# Generates the storage services date string
```
Comment (Shan1024) [inline azurecommons/azurecommons.bal]:
```suggestion
# Generates the storage services date string.
```

File: azurecommons/azurecommons.bal [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    value = headers["Content-MD5"];
+    if (value is string) {
+        contentMD5 = value;
+    }
+    string date = "";
+    value = headers["Date"];
+    if (value is string) {
+        date = value;
+    }
+    string stringToSign = verb.toUpper() + "\n" +
```
Comment (Shan1024) [inline azurecommons/azurecommons.bal]: I think it is better to merge these concatenations into a single (or 2) lines. Or use string template for this.

Comment (lafernando) [reply azurecommons/azurecommons.bal]: Just added those to each line for it to be a bit extra clearer.

**Pull request merged**